src/crawler.js

```diff
@@ -71,22 +75,43 @@ function normalizeIpp(ip, port) {
 
 function Crawler(maxRequests, logger) {
   EventEmitter.call(this);
-  this.maxRequests = maxRequests ? maxRequests : 30;
+  
+ 
+  // maxRequests checks
+  maxRequests = maxRequests ? maxRequests : 30;
+  if (!maxRequests || maxRequests < 1) {
+    throw new Error("Invalid max requests");
+  }
+  check.assert.number(maxRequests, "Invalid max requests");
```

Comment:
This should be moved to the line before ``maxRequests < 1``, but after ``maxRequests = ...``.

First check makes sure that you've got some default value

Then the code should check if the value passed in is in fact a number.

Finally, check if the number is within the valid range.